.gitattributes

```diff
@@ -0,0 +1,2 @@
+# Always use LF line endings even on Windows.
+* text eol=lf
```

Comment:
Aha thank you for raising this point! @tsibley had a similar [post-merge review in mpox](https://github.com/nextstrain/mpox/pull/47#discussion_r894036742):

> I don't think we want to force the text attribute here. While we don't yet have any non-text (binary) files in the repo at the moment, we could easily have them in the future (.gz, .png, etc.) and forcing text mode will corrupt them.
> 
> I think it should be enough to set eol=lf and let git automatically detect text files per it's normal config? Or, if we want to force automatic detection regardless of config, we could specify text=auto.

Reply:
That's a good question. Looking more at [the docs you linked above](https://git-scm.com/docs/gitattributes#_checking_out_and_checking_in), @jameshadfield, this implementation (`* text eol=lf`) will treat all files as text and try to apply the LF EOL conversion, but if we used `* text=auto eol=lf` then Git would first determine whether the file is text or binary and only apply the conversion to the text files. 

> When text is set to "auto", Git decides by itself whether the file is text or binary. If it is text and the file was not already in Git with CRLF endings, line endings are converted on checkin and checkout as described above. Otherwise, no conversion is done on checkin or checkout.

But it also looks like that "auto" mode is not always reliable at detecting binary files, since [GitHub docs recommend specifying binary file types](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#example) to prevent accidental treatment as text by Git.

For pathogen workflows, we don't usually include binary files, but it seems important to be aware of the potential binary-interpreted-as-text issue...

Reply:
Updated to `* text=auto eol=lf` e70d7344573484db9026ae964ce90e7df92759ba

Reply:
> For pathogen workflows, we don't usually include binary files,

I think compressed example data aren't too uncommon ([e.g.](https://github.com/nextstrain/ncov/tree/master/data)). 

I'm trying to understand the mechanism by which binary files could become corrupted with `* text` - if a `LF`-like sequence appears in a binary file then no modification would be done, but if a `CRLF`-like sequence appeared then git would change this to `LF` when it added it to the index - is this right?

Reply:
> if a `LF`-like sequence appears in a binary file then no modification would be done, but if a `CRLF`-like sequence appeared then git would change this to `LF` when it added it to the index - is this right?

Sounds right, in line with the warning shown when adding a PNG file under `* text eol=lf`:

```
$ git add test.png 
warning: in the working copy of 'test.png', CRLF will be replaced by LF the next time Git touches it
```

[This](https://github.com/desktop/desktop/issues/5323#issuecomment-410220449) explanation of `* text eol=lf` is useful:

> this rule essentially says "for all files attempt to do line ending conversion and ensure the line endings are set to `LF`". That's why Git has gone and changed the line endings inside the [PNG file signature](http://www.libpng.org/pub/png/spec/1.2/PNG-Rationale.html#R.PNG-file-signature) - it's spotted the `CRLF` at the start of the file and applied this filter rule to it.